bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/client/AbstractKNXClient.java

```diff
@@ -299,6 +300,11 @@ private void readNextQueuedDatapoint() {
                     logger.warn("Giving up reading datapoint {}, the number of maximum retries ({}) is reached.",
                             datapoint.getDatapoint().getMainAddress(), datapoint.getLimit());
                 }
+            } catch (KnxRuntimeException e) {
```

Comment:
I think you are hiding a bug here. `KnxRuntimeException` is inherited from `RuntimeException` by design. In the linked issue, `KNXIllegalArgumentException` is thrown, so this seems to suggest that `read()` is called with an invalid parameter, `datapoint.getDatapoint()`, in this scenario?

Reply:
Currently, all kind of KnxRuntimeExceptions will end the scheduled runnable, so all scheduled reads will not be executed after the exception caused by one broken item (as shown in #7239). This effectivly breaks all scheduled reads, not only for the problematic item.

The approach here is targeted to improve the behaviour in case a problem occurs.
I think using the warning for the log this will be sufficient to show that there is an issue.

Remark: KnxSecureException is derived from KnxRuntimeException. AFAIR I was able to trigger it by resetting the secure IP Interface (work for #12434), but let me double check this....

Reply:
I don't know any details here, so something just seemed wrong when catching a RuntimeException as this usually hides deeper problems. So now checked the method throwing the exception:

```
	/**
	 * Sets the data array with KNX datapoint type items for translation.
	 * <p>
	 * The <code>data</code> array contains at least one DPT item, the new item(s) will
	 * replace any other items set in the translator before.<br>
	 * The number of items (KNX data values) for translation in <code>data</code> is
	 * inferred from the length of the usable <code>data</code> range:<br>
	 * <code>items = (data.length - offset) / (length of KNX data type)</code>
	 * <p>
	 * In general, the KNX data type width is implicitly known in the context where a
	 * translator is invoked (e.g., by appropriate DP configuration). Therefore,
	 * <code>data.length</code> will satisfy the minimum acceptable length. If this is
	 * not the case, {@link KNXIllegalArgumentException} has to be caught and handled in
	 * the caller's context.
	 *
	 * @param data byte array containing KNX DPT item(s)
	 * @param offset offset into <code>data</code> from where to start, 0 &lt;= offset
	 *        &lt; <code>data.length</code>
	 * @throws KNXIllegalArgumentException if <code>data.length</code> - offset &lt;
	 *         data type width of this DPTXlator
	 */
	public void setData(final byte[] data, final int offset)
	{
		if (offset < 0 || offset > data.length)
			throw new KNXIllegalArgumentException("illegal offset " + offset);
		final int size = Math.max(1, getTypeSize());
		final int length = (data.length - offset) / size * size;
		if (length == 0)
			throw new KNXIllegalArgumentException("DPT " + dpt.getID() + " " + dpt.getDescription() + ": data length "
					+ (data.length - offset) + " < required datapoint type width " + size);
		this.data = new short[length];
		for (int i = 0; i < length; ++i)
			this.data[i] = ubyte(data[offset + i]);
	}
```

So if in your case the KNX data type width is not known in your context, I guess it's okay to catch and handle the exception. However, if you could somehow validate before calling, that might be preferable.